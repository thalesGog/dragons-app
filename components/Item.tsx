import styled from 'styled-components'

const Item = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  border-radius: 0.125rem;
  background: #fff;
  margin-top: 1.5rem;
  padding: 1rem 1.5rem;
  max-height: 100px;
  font-size: 1.25rem;
  text-align: start;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  cursor: pointer;
  transition: all 300ms ease 0s;
  a {
    font-size: 0.9rem;
    text-decoration: none;
    color: ${({theme}) => theme.colors.secondary};
    margin-top: 8px;
  }
  hr {
    width: 100%;
  }
  &:hover {
    box-shadow: 4px 8px ${({theme}) => theme.colors.primary};
  }
`

export default Item
