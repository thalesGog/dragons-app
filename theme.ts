import {DefaultTheme} from 'styled-components'

const theme: DefaultTheme = {
  fontFamily: 'Roboto, sans-serif',
  backgroundColor: '#fff0e5',
  colors: {
    primary: '#10162f',
    secondary: '#3a10e5',
  },
}

export default theme
