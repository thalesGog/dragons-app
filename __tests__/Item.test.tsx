import React from 'react'
import {ThemeProvider} from 'styled-components'
import theme from '../theme'
import {render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import Item from '../components/Item'

test('Render Item component', async () => {
  const {container} = render(
    <ThemeProvider theme={theme}>
      <Item />
    </ThemeProvider>,
  )
  expect(container).toMatchSnapshot()
})
