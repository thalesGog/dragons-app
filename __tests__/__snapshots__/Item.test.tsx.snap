// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Item component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: 0.125rem;
  background: #fff;
  margin-top: 1.5rem;
  padding: 1rem 1.5rem;
  font-size: 1.25rem;
  text-align: start;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  -webkit-transition: all 300ms ease 0s;
  transition: all 300ms ease 0s;
}

.c0 a,
.c0 span {
  font-size: 0.9rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3a10e5;
  margin-top: 8px;
  cursor: pointer;
}

.c0 a:hover,
.c0 span:hover {
  font-weight: 600;
}

.c0 hr {
  width: 100%;
}

.c0:hover {
  box-shadow: 4px 8px #10162f;
}

<div>
  <div
    class="c0"
  />
</div>
`;
